Refactor Ray Cluster/AppWrapper creation (#650)
project-codeflare/codeflare-sdk: docs/ray_cluster_interaction.md

# Ray Cluster Interaction

The CodeFlare SDK offers multiple ways to interact with Ray Clusters including the below methods.

## get_cluster()

> The `get_cluster()` command is used to initialise a `Cluster` object from a pre-existing Ray Cluster/AppWrapper. <br>

Reviewer: I think use `method` or `function` here instead of `command`?

Below is an example of it's usage:
```
from codeflare_sdk import get_cluster
cluster = get_cluster(cluster_name="raytest", namespace="example", is_appwrapper=False, write_to_file=False)
-> output: Yaml resources loaded for raytest

cluster.status()
-> output:
                    🚀 CodeFlare Cluster Status 🚀

 ╭─────────────────────────────────────────────────────────────────╮
 │   Name                                                          │
 │   raytest                                           Active ✅   │
 │                                                                 │
 │   URI: ray://raytest-head-svc.example.svc:10001                 │
 │                                                                 │
 │   Dashboard🔗                                                   │
 │                                                                 │
 ╰─────────────────────────────────────────────────────────────────╯
(<CodeFlareClusterStatus.READY: 1>, True)

cluster.down()

cluster.up() # This method will create an exact copy of the retrieved Ray Cluster only if the Ray Cluster has been previously deleted.
```

> These are the parameters the `get_cluster()` method accepts:

Reviewer: Update the use of  `method` here if using `function` above on line 6

* `cluster_name: str # Required` -> The name of the Ray Cluster.

> * `namespace: str # Default: "default"` -> The namespace the Cluster.

Reviewer: ```suggestion
* `namespace: str # Default: "default"` -> The namespace of the Ray Cluster.
```

> * `is_appwrapper: bool # Default: False` -> The function will attempt to retrieve an AppWrapper instead of a Ray Cluster.

Reviewer: ```suggestion
* `is_appwrapper: bool # Default: False` -> When set to `True` the function will attempt to retrieve an AppWrapper instead of a Ray Cluster.
```

> * `write_to_file: bool # Default: False` -> The Ray Cluster/AppWrapper will be written to a file similar to how it is done in `ClusterConfiguration`.

Reviewer: ```suggestion
* `write_to_file: bool # Default: False` -> When set to `True` the Ray Cluster/AppWrapper will be written to a file similar to how it is done in `ClusterConfiguration`.
```

## list_all_queued()

> The `list_all_queued()` returns (and prints by default) a list of all currently queued-up Ray Clusters in a given namespace.

Reviewer: function or method, whichever you are going with
```suggestion
The `list_all_queued()` function returns (and prints by default) a list of all currently queued Ray Clusters in a given namespace.
```

It accepts the following parameters:
* `namespace: str # Required` -> The namespace you want to retrieve the list from.
* `print_to_console: bool # Default: True` -> Allows the user to print the list to their console.

> * `appwrapper: bool # Default: False` -> Allows the user to list queued AppWrappers.

Reviewer: ```suggestion
* `appwrapper: bool # Default: False` -> Setting to `True` allows the user to list queued AppWrappers.
```

## list_all_clusters()
The `list_all_clusters()` function will return a list of detailed descriptions of Ray Clusters to the console by default. It accepts the following parameters:
* `namespace: str # Required` -> The namespace you want to retrieve the list from.
* `print_to_console: bool # Default: True` -> A boolean that allows the user to print the list to their console.

<br>
NOTE: The following methods require a `Cluster` object to be initialised see [Cluster Configuration](./cluster-configuration.md)

## cluster.up()
The `cluster.up()` method creates a Ray Cluster in the given namespace.

## cluster.down()

> The `cluster.down()` method will delete the Ray Cluster in the given namespace.

Reviewer: ```suggestion
The `cluster.down()` method deletes the Ray Cluster in the given namespace.
```

## cluster.status()

> The `cluster.status()` method will print out a status of the Ray Cluster's state with a link to the Ray Dashboard.

Reviewer: ```suggestion
The `cluster.status()` method prints out the status of the Ray Cluster with a link to the Ray Dashboard.
```

## cluster.details()

> The `cluster.details()` method will print out a detailed description of the Ray Cluster's status, worker resources and a link to the Ray Dashboard.

Reviewer: ```suggestion
The `cluster.details()` method prints out a detailed description of the Ray Cluster's status, worker resources and a link to the Ray Dashboard.
```

## cluster.wait_ready()

> The `cluster.wait_ready()` method waits for requested cluster to be ready, up to an optional timeout and checks every 5 seconds. It accepts the following parameters:

Reviewer: ```suggestion
The `cluster.wait_ready()` method waits for the requested cluster to be ready, up to an optional timeout and checks every 5 seconds. It accepts the following parameters:
```
